### DISCUSSION TITLE
negative video color playback

### DISCUSSION THREAD

**Discussion by OmiChrome** [hooke007/mpv_PlayKit#281] [z_ENG_discuss]
i used  `o cycle-values vf negate "" ` in my input.conf 
it worked well but after the new update of mpv it dosen't seem to work 

here is breakdown what it does 
`o cycle-values vf negate "" ` = negative the color , as black to white , or white to black - vice versa 
its so usefull in place where video is soo much white and ur using screen in night | save eyes from burning

commnad i was trying 
`mpv --vf=negate  $file `
but there is error 
```
Using hardware decoding (d3d11va).
[ffmpeg] Impossible to convert between the formats supported by the filter 'mpv_src_default_in' and the filter 'auto_scale_0'
[lavfi] failed to configure the filter graph
Disabling filter negate.00 because it has failed.
```
i dont know if its error with `ffmpeg` or `mpv` 
im just looking for **FIX**
i really want to use that feature  :( help me

**Comment by hooke007** [comment]
native hwdec doesn‘t work with lavfi filters. It is not a bug.

**Reply by OmiChrome** [reply]
THX !! i i got  what it means , and what i have to do

**Reply by OmiChrome** [reply]
but why its  it doesn't support :( , is there any other work around ? 
since im seeing so many people sticking to native hwdec ,

**Reply by hooke007** [reply]
mpv doesn't know the fmt generated by lavfi-filters.
There is no real workaround. Because most lavfi-filter only works on CPU. You have to force the final decoding to return to CPU.

**Reply by OmiChrome** [reply]
okay  i found kinda of fix 
`hwdec=auto-safe-copy  ` in mpv.conf for hardware decoding at some point 

with this [invert.glsl ](https://github.com/azimut/dotfiles/blob/master/homedir/.config/mpv/shaders/invert.glsl) shader 


from => https://github.com/mpv-player/mpv/issues/9338

i want your thoughts @hooke007 
some info 

CPU :  Core i3-1115G4 
RAM : 8gb 
its a laptop 

my [config](https://github.com/omaru-ok/stuff/blob/main/Personal/portable_config.zip)

**Reply by hooke007** [reply]
`auto-safe-copy` is what I said —— "You have to force the final decoding to return to CPU"

**Reply by OmiChrome** [reply]
i think it suggest its good
